API/API/Services/Classes/ProductService.cs

```diff
@@ -0,0 +1,80 @@
+﻿using API.General.Classes.Configure;
+using API.Models;
+using API.Repositories.Interfaces;
+using API.Services.Interfaces;
+
+// <summary>
+// Developer....: Karla Ramos Benitez       USER ID: XKRB
+// </summary>
+namespace API.Services.Classes;
+
+public class ProductService : IProductService
+{
+    /// <summary>
+    /// User Model
+    /// </summary>
+
+    //private readonly ProductModel? _product = null;
+
+    /// <summary>
+    /// To manage products's table data
+    /// </summary>
+    private readonly IProductRepository _productRepository;
+
+    /// <summary>
+    /// Parameters are passed via dependency injection to manage tables data
+    /// </summary>
+    public ProductService(IProductRepository productRepository) => _productRepository = productRepository;
+
+    /// <summary>
+    /// Get product
+    /// </summary>
+    /// <param name="product"> product id </param>
+    /// <returns> product id, product name and product price </returns>
+    public async Task<ProductModel> GetProduct(int product)
+    {
+        ProductModel productModel = await _productRepository.GetProduct(product);
+        if (productModel == null)
+        {
+            throw new ExceptionClass("Product was not found");
+        }
+        else
+        {
+            return productModel;
+        }
+
+    }
+
+    /// <summary>
+    /// Create product
+    /// </summary>
+    /// <param name="product"> Create product </param>
+    public async Task<ProductModel> CreateProduct(ProductModel product)
+    {
+        ProductModel productModel = await _productRepository.GetProduct(product.IdProduct);
+        if (productModel != null)
```

Comment:
This validation is already done in the Repository.

Reply:
No, I have the validation of update product and delete product in the Repository and the other validation (Get & Post) in the Service. should  I put all the validations in the Repository?